coordinator/internal/transitengine/transitengine_test.go

```diff
@@ -0,0 +1,167 @@
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/edgelesssys/contrast/coordinator/internal/authority"
+	"github.com/edgelesssys/contrast/coordinator/internal/seedengine"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCryptoAPICyclic(t *testing.T) {
+	type testData struct {
+		name           string
+		Plaintext      string `json:"plaintext"`
+		AssociatedData string `json:"associated_data,omitempty"`
+		Version        int    `json:"key_version"`
+		expStatus      int
+	}
+	mocks := map[string]testData{
+		"positive": {
+			Plaintext:      "vpIhKQhFuGwLv5B/XLYr960uZQ==",
+			name:           "autounseal",
+			AssociatedData: "Xv0nZLWkSan+vdWrH2LrGP8TU/Qg1+ZX7vldWMbxKTk=",
+			Version:        2,
+			expStatus:      200,
+		},
+		"special_char_name": {
+			Plaintext:      "lT3rQGMlxq680DdSKfIYYcfyCfMnP9ikxaO5b0mGRKRl4qNL3W9xkSW3QmaMwozCRfNMZhhDCbYokn6KEiGotlVInKt66QjBgXR2Nk9hIcez0LYt8W5pxD0lwTxC",
+			name:           "thi$$hoU_ld+*work",
+			AssociatedData: "HribV+LFZspJpAauFf643A1HKbj1VlQWVhAKFDJqdZg=",
+			Version:        2000,
+			expStatus:      200,
+		},
+		"f_empty_name": {
+			Plaintext:      "vpIh",
+			name:           "",
+			AssociatedData: "vpIh",
+			expStatus:      404,
+		},
+		"f_name_with_backsl": {
+			Plaintext:      "vpIh",
+			name:           "wrong/url",
+			AssociatedData: "vpIh",
+			expStatus:      404,
+		},
+		"f_bad-request-nob64plain": {
+			Plaintext:      "thi$$hoU_ld+*notwork",
+			name:           "autounseal",
+			AssociatedData: "vpIh",
+			expStatus:      400,
+		},
+		"f_bad-request-nob64AD": {
+			Plaintext:      "HribV+LFZspJpAauFf643A1HKbj1VlQWVhAKFDJqdZg=",
+			name:           "autounseal",
+			AssociatedData: "thi$$hoU_ld+*notwork",
+			expStatus:      400,
+		},
+		"f_neg_version": {
+			Plaintext:      "lT3rQGMlxq680DdSKfIYYcfyCfMnP9ikxaO5b0mGRKRl4qNL3W9xkSW3QmaMwozCRfNMZhhDCbYokn6KEiGotlVInKt66QjBgXR2Nk9hIcez0LYt8W5pxD0lwTxC",
+			name:           "thi$$hoU_ld+*work",
+			AssociatedData: "HribV+LFZspJpAauFf643A1HKbj1VlQWVhAKFDJqdZg=",
+			Version:        -2000,
+			expStatus:      400,
+		},
+	}
+
+	t.Run("encrypt-decrypt handler", func(t *testing.T) {
+		fakeStateAuthority, _ := NewFakeSeedEngineAuthority()
+		mux := NewTransitEngineAPI(fakeStateAuthority, slog.Default())
+
+		for testname, entry := range mocks {
+			t.Run(testname, func(t *testing.T) {
+				var ciphertext, receivedPlaintext string
+				t.Run("encryption request handling", func(t *testing.T) {
+					require := require.New(t)
+					jsonBody, err := json.Marshal(entry)
+					require.NoError(err)
+					req := httptest.NewRequest(http.MethodPut, "/v1/transit/encrypt/"+entry.name, bytes.NewReader(jsonBody))
+					req.Header.Set("Content-Type", "application/json")
+					rec := httptest.NewRecorder()
+					mux.ServeHTTP(rec, req)
+					res := rec.Result()
+					require.Equal(entry.expStatus, res.StatusCode)
+
+					if res.StatusCode != http.StatusOK {
+						return
```

Comment:
Please leave a TODO to implement Vault JSON error bodies: https://developer.hashicorp.com/vault/api-docs#error-response. Does not need to happen in this PR.